ols/app/endpoints/feedback.py

```diff
@@ -1,33 +1,197 @@
+    logger.info(f"'{len(feedbacks)}' feedbacks found")
+    return feedbacks
+
+
+def store_feedback(user_id: str, feedback: dict) -> None:
+    """Store feedback in the local filesystem.
+
+    Args:
+        user_id: The user ID (UUID).
+        feedback: The feedback to store.
+    """
+    # ensures storage path exists
+    storage_path = Path(config.ols_config.user_data_collection.feedback_storage)
+    if not storage_path.exists():
+        logger.debug(f"creating feedback storage directories '{storage_path}'")
+        storage_path.mkdir(parents=True)
+
+    data_to_store = {"user_id": user_id, **feedback}
+
+    # stores feedback in a file under unique uuid
+    feedback_file_path = storage_path / f"{get_suid()}.json"
+    with open(feedback_file_path, "w", encoding="utf-8") as feedback_file:
+        json.dump(data_to_store, feedback_file)
+
+    logger.info(f"feedback stored in '{feedback_file_path}'")
+
+
+def remove_feedback(feedback_id: str) -> None:
+    """Remove feedback from the local filesystem.
+
+    Args:
+        feedback_id: The feedback ID (UUID).
+    """
+    storage_path = Path(config.ols_config.user_data_collection.feedback_storage)
+    feedback_file = storage_path / f"{feedback_id}.json"
+    if feedback_file.exists():
+        feedback_file.unlink()
+        if not feedback_file.exists():
+            logger.info(f"feedback file '{feedback_file}' removed")
+        else:
+            logger.error(f"feedback file '{feedback_file}' failed to remove")
+    else:
+        logger.error(f"feedback file '{feedback_file}' not found")
+
+
+@router.get("/status")
+def feedback_status() -> StatusResponse:
+    """Handle feedback status requests.
+
+    Returns:
+        Response indicating the status of the feedback.
+    """
+    logger.info("feedback status request received")
+    feedback_status = get_feedback_status()
+    return StatusResponse(functionality="feedback", status={"enabled": feedback_status})
+
+
+# TODO: OLS-136 implements the collection mechanism - revisit the need
+# of this endpoint
+# If endpoint stays in place, it needs to be properly secured - OLS-404
+@router.get("/list")
+def get_user_feedbacks(
+    feedback_is_enabled: Any = Depends(feedback_is_enabled),
+    auth: Any = Depends(auth_dependency),
+) -> FeedbacksListResponse:
+    """Handle feedback listing requests.
+
+    Args:
+        feedback_is_enabled: The feedback handler (FastAPI Depends) that
+            will handle feedback status checks.
+        auth: The Authentication handler (FastAPI Depends) that will
+            handle authentication Logic.
+
+    Returns:
+        Response containing the list of feedbacks.
+    """
+    logger.info("feedback list request received")
+
+    feedbacks = list_feedbacks()
+
+    return FeedbacksListResponse(feedbacks=feedbacks)
+
+
 @router.post("")
-def user_feedback(
-    feedback_request: FeedbackRequest, auth: Any = Depends(auth_dependency)
+def store_user_feedback(
```

Comment:
should it return 201 Created?

Reply:
I believe that this question is related to https://github.com/openshift/lightspeed-service/pull/538#discussion_r1530896191
It comes from our schema generator

Reply:
schema is generated from our sources, not vice versa;)